tests/test_database_types.py

```diff
@@ -2,33 +2,117 @@
 import unittest
 import time
 import logging
+import re
+import math
+import datetime
 from decimal import Decimal
+from prestodb.exceptions import PrestoUserError
 
 from parameterized import parameterized
 
 from data_diff import databases as db
 from data_diff.diff_tables import TableDiffer, TableSegment
-from .common import CONN_STRINGS
+from .common import CONN_STRINGS, N_SAMPLES
 
 
-logging.getLogger("diff_tables").setLevel(logging.ERROR)
-logging.getLogger("database").setLevel(logging.WARN)
-
-CONNS = {k: db.connect_to_uri(v) for k, v in CONN_STRINGS.items()}
+CONNS = {k: db.connect_to_uri(v, 1) for k, v in CONN_STRINGS.items()}
 
 CONNS[db.MySQL].query("SET @@session.time_zone='+00:00'", None)
 
-TYPE_SAMPLES = {
-    "int": [127, -3, -9, 37, 15, 127],
-    "datetime_no_timezone": [
-        "2020-01-01 15:10:10",
-        "2020-02-01 9:9:9",
-        "2022-03-01 15:10:01.139",
-        "2022-04-01 15:10:02.020409",
-        "2022-05-01 15:10:03.003030",
-        "2022-06-01 15:10:05.009900",
-    ],
-    "float": [
+class PaginatedTable:
+    # We can't query all the rows at once for large tables. It'll occupy too
+    # much memory.
+    RECORDS_PER_BATCH = 1000000
+
+    def __init__(self, table, conn):
+        self.table = table
+        self.conn = conn
+
+    def __iter__(self):
```

Comment:
I think this kind of mutable iterator where you always return `self` in `__iter__` isn't thread safe. And these tests are much better when they can be run in parallel.

To make it thread-safe, `__iter__` should return a new `PaginatedTableIter` which keeps all the mutable attributes contained.

If you want to test it, use a package called `unittest-parallel`.